alf/trainers/on_policy_trainer.py

```diff
@@ -124,15 +126,19 @@ def train_():
 
             if eval_env is not None and (iter + 1) % eval_interval == 0:
                 with tf.summary.record_if(True):
-                    eager_compute(metrics=eval_metrics,
-                                  environment=eval_env,
-                                  state_spec=algorithm.predict_state_spec,
-                                  action_fn=algorithm.greedy_predict,
-                                  num_episodes=num_eval_episodes,
-                                  train_step=global_step,
-                                  summary_writer=eval_summary_writer,
-                                  summary_prefix="Metrics")
+                    eager_compute(
+                        metrics=eval_metrics,
+                        environment=eval_env,
+                        state_spec=algorithm.predict_state_spec,
+                        action_fn=algorithm.greedy_predict,
+                        num_episodes=num_eval_episodes,
+                        train_step=global_step,
+                        summary_writer=eval_summary_writer,
+                        summary_prefix="Metrics")
                     metric_utils.log_metrics(eval_metrics)
+            if iter == 0:
```

Comment:
it's better to move  line 139-141   before the loop L144 ?

Reply:
The functions need be called in order to get the operative configurations from gin. So we need to do the summary after the major functions